janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/CacheEdge.java

```diff
@@ -87,9 +87,12 @@ private synchronized InternalRelation update() {
         copyProperties(copy);
         copy.remove();
 
-        StandardEdge u = (StandardEdge) tx().addEdge(getVertex(0), getVertex(1), edgeLabel());
-        if (type.getConsistencyModifier()!=ConsistencyModifier.FORK) u.setId(super.longId());
-        u.setPreviousID(super.longId());
+        Long id = null;
+        if (type.getConsistencyModifier() != ConsistencyModifier.FORK) {
+            id = longId();
+        };
```

Comment:
Nitpick. Don't assign `null` to the variable when it isn't needed to be assigned. I.e. change the code to:
```
Long id;
if (type.getConsistencyModifier() != ConsistencyModifier.FORK) {
  id = longId();
} else {
  id = null;
}
```
I'm also OK with ternary operator if `if{}else{}` isn't suitable. I.e.:
```
Long id = (type.getConsistencyModifier() != ConsistencyModifier.FORK) ? longId() : null;
```

Reply:
Done, but may I know why?

Reply:
Thanks! Assigning value to a variable which we know can be changed in the next operation and which isn't used before that operation is a redundant operation.

Reply:
I see. Are you speaking from a code style perspective? Then that would be make some sense to me. I would be surprised and interested if there is actually difference in runtime performance.

Reply:
I don't know if JIT optimizes such cases or not, but from bytecode perspective there is a very tiny difference. It's almost not noticeable and most likely won't affect performance in real life but there is a difference from operations amount. I don't want to check the byte code of this class but lets see the bytecode difference in much smaller use case:
```
/*
0: aconst_null
1: astore_1
2: aload_0
3: arraylength
4: iconst_5
5: if_icmple     13
8: iconst_5
9: invokestatic  #2                  // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
12: astore_1
13: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
16: aload_1
17: invokevirtual #4                  // Method java/io/PrintStream.println:(Ljava/lang/Object;)V
20: return
 */
public static void main(String[] args) {

    Integer s = null;

    if(args.length > 5){
        s = 5;
    }

    System.out.println(s);
}

/*
0: aload_0
1: arraylength
2: iconst_5
3: if_icmple     14
6: iconst_5
7: invokestatic  #2                  // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
10: astore_1
11: goto          16
14: aconst_null
15: astore_1
16: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
19: aload_1
20: invokevirtual #4                  // Method java/io/PrintStream.println:(Ljava/lang/Object;)V
23: return
 */
public static void main(String[] args) {

    Integer s;

    if(args.length > 5){
        s = 5;
    } else {
        s = null;
    }

    System.out.println(s);
}
```
In the first main method there are 13 operations in the main method. In the second main method we have 14 operations in the main method. We might assume that the second method will be executed slower than the first method because it has more operations but if we read them we clearly see that it's the opposite way. 
In case `args.length > 5` is `false` than we will execute 10 operations in the first method and 10 operations in the second method. So it's a tie when the if statement is `false`.
In case `args.length > 5` is `true` than we will execute 13 operations in the first method and 12 operations in the second method (notice `goto 16` operation). So in case `if` statement is `true` the second method will perform 1 operation less than the first method.

In conclusion: JIT compiler *may* optimize such use cases but I'm not sure about that. Comparing the performance for such a small difference for modern apps is a waste of time. In the real world most likely we won't see any performance difference in such a small use cases but it doesn't mean that we shouldn't care about redundant operations at all because they *may* decrease the performance and they just increase the code complexity. 
That said, I don't push to use such style and I'm open for other opinions in case you have arguments to use the opposite style, I'm open to discuss that

Reply:
Very interesting analysis! I am also not sure if JIT compiler will optimize this. I agree that comparing the performance for such a small difference is not valuable, but since you have the byte code pasted, I'd also like to add some points:

> In case args.length > 5 is true than we will execute 13 operations in the first method and 12 operations in the second method (notice goto 16 operation). So in case if statement is true the second method will perform 1 operation less than the first method.

I agree the second method seems to save one operation, but I doubt if it will be actually faster than the first method. Reasons are:

1) goto operation MIGHT be slower than a normal operation since a jump probably does not leverage cache spatial locality well.
2) CPU branch prediction MIGHT suffer more on goto operation when branch prediction fails. This is just my conjecture though.

Reply:
Very good points!